src/commodore-helm/index.ts

```diff
@@ -0,0 +1,243 @@
+  });
+  const componentKey: string = componentKeyFromName(dep.groupName);
+  const charts: Map<string, string> = defaults.parameters[componentKey].charts;
+  if (!charts) {
+    return null;
+  }
+
+  let deps: PackageDependency[] = Object.entries(charts).map(
+    ([chartName, chartVersion]) => {
+      let d = config.baseDeps.find((d: PackageDependency) => {
+        return d.propSource == chartName;
+      });
+      let res: PackageDependency = {
+        depName: d.depName,
+        currentValue: chartVersion,
+      };
+      return res;
+    }
+  );
+
+  return { deps };
+}
+
+export async function extractAllPackageFiles(
+  _config: ExtractConfig,
+  packageFiles: string[]
+): Promise<PackageFile[] | null> {
+  let fileContents: Map<string, string> = new Map();
+  let componentName: string | null = null;
+  let defaults_file: string | null = null;
+
+  for (const file of packageFiles) {
+    const content = await readLocalFile(file, 'utf8');
+    const key = path.parse(file).name;
+    fileContents.set(key, content);
+    if (key != 'defaults') {
+      componentName = key;
+    } else {
+      defaults_file = file;
+    }
+  }
+
+  if (!fileContents.has('defaults')) {
+    logger.error('Component repository has no `class/defaults.ya?ml`');
+    return null;
+  }
+
+  if (!componentName) {
+    logger.error(
+      { packageFiles },
+      'Unable to identify component name from package files'
+    );
+    return null;
+  }
+
+  const defaults: any = yaml.load(fileContents.get('defaults') ?? '');
+  const component: any = yaml.load(fileContents.get(componentName) ?? '');
+
+  if (
+    defaults === undefined ||
+    defaults === null ||
+    defaults.parameters === undefined ||
+    defaults.parameters === null ||
+    component === undefined ||
+    component === null ||
+    component.parameters === undefined ||
+    component.parameters === null
+  ) {
+    return null;
+  }
+
+  const deps: PackageDependency[] = extractHelmChartDependencies(
+    componentName,
+    defaults.parameters,
+    component.parameters
+  );
+
+  if (deps.length == 0) {
+    return null;
+  }
+
+  return [{ packageFile: defaults_file, datasource: 'helm', deps }];
+}
+
+interface KapitanDependency {
+  type: string;
+  source: string;
+  output_path: string;
+}
+interface KapitanHelmDependency extends KapitanDependency {
+  chart_name: string;
+  version: string;
+}
+
+function extractHelmChartDependencies(
+  componentName: string,
+  defaults: Record<string, any>,
+  componentParams: Record<string, any>
+): PackageDependency[] {
+  if (
+    componentName === '' ||
+    defaults === undefined ||
+    defaults === null ||
+    typeof defaults !== 'object' ||
+    componentParams === undefined ||
+    componentParams === null ||
+    typeof componentParams !== 'object'
+  ) {
+    return [];
+  }
+
+  const componentKey: string = componentKeyFromName(componentName);
+
+  const chartVersions: Map<string, string> = defaults[componentKey].charts;
```

Comment:
```suggestion
  const chartVersions: Map<string, string> = defaults[componentKey]?.charts;
```
As `defaults` might not have an entry for the component, right?

Reply:
Fixed by adding an explicit null check for `defaults[componentKey]` to provide a better error message for that case, cf. https://github.com/projectsyn/commodore-renovate/pull/17/commits/928112f10fa54992651437116fbd6e685ed67ec3